wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/WebSocketSettings.java

```diff
@@ -245,6 +246,16 @@ public Executor getSendPayloadExecutor()
 		return sendPayloadExecutor;
 	}
 
+	/**
+	 * Allows to configure {@link org.apache.wicket.protocol.ws.api.IWebSocketSession}
+	 *
+	 * @param webSocketSession The {@link org.apache.wicket.protocol.ws.api.IWebSocketSession}
+	 */
+	public void configureSession(IWebSocketSession webSocketSession)
```

Comment:
I wonder whether we need more abstractions here.
It feels weird a Settings class to `do` something. Usually the Settings classes have setters/getters for the do-ers.
E.g. 
```java
public void setWebSocketSessionConfigurer(IWebSocketSessionConfigurer configurer) 
{
   this.webSocketSessionConfigurer = Args.nonNull(configurer, "configurer");
}
```

The default configurer would do nothing in its `#configure()` method.

Reply:
Yes...you are right. We are somehow "forcing" the settings class to acta as a listener. Thus we might need to introduce a Listener/Configurer  that is set in Setting and laster on called. This might  be more consistent.

Reply:
Done!